tests/baselines/reference/parentheses.js

```diff
@@ -0,0 +1,18 @@
+//// [parentheses.ts]
+declare const o1: ((...args: any[]) => number);
+declare const o2: { b: (...args: any[]) => number };
+declare const o3: { b: ((...args: any[]) => (...args: any[]) => number) };
+declare const o4: { b: ((...args: any[]) => { c: (...args: any[]) => number } ) };
+
+(o1)(o1 ?? 1);
+(o2?.b)(o1 ?? 1);
+(o3?.b())(o1 ?? 1);
+(o4?.b().c)(o1 ?? 1);
+
+
+//// [parentheses.js]
+var _a;
+(o1)(o1 !== null && o1 !== void 0 ? o1 : 1);
+(o2 === null || o2 === void 0 ? void 0 : o2.b).call(o2, o1 !== null && o1 !== void 0 ? o1 : 1);

```

Comment:
Nit: this doesn’t need a `.call`:

```js
o2 === null || o2 === undefined ? void 0 : o2.b(o1 ?? 1);

Reply:
But that wouldn't throw an error if `o2` is nullish.

Reply:
Ohh, it's wrapped in parens. Never mind, forgot that made the call non-optional.